### PR TITLE
[BEAM-2829] Add an option for dataflow job labels.

### PR DESCRIPTION
**Pull request by aaltay**
R: @zongweiz @charlesccychen

### CODE REVIEW

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 1]
```diff
@@ -374,6 +374,13 @@ def _add_argparse_args(cls, parser):
     parser.add_argument('--template_location',
                         default=None,
                         help='Save job to specified local or GCS location.')
+    parser.add_argument(
+        '--label', '--labels',
+        dest='labels',
+        action='append',
+        default=None,
+        help='Labels that will be applied to the billing records for this job. '
```
Comment (charlesccychen) [inline sdks/python/apache_beam/options/pipeline_options.py]: Is "labels" a general purpose Dataflow concept, or just for billing?

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py [hunk 2]
```diff
@@ -363,6 +363,17 @@ def __init__(self, options):
       self.proto.type = dataflow.Job.TypeValueValuesEnum.JOB_TYPE_STREAMING
     else:
       self.proto.type = dataflow.Job.TypeValueValuesEnum.JOB_TYPE_BATCH
+
+    # Labels
```
Comment (charlesccychen) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: nit: period after

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py [hunk 3]
```diff
@@ -363,6 +363,17 @@ def __init__(self, options):
...
+    if self.google_cloud_options.labels:
+      self.proto.labels = dataflow.Job.LabelsValue()
+      for label in self.google_cloud_options.labels:
+        label = label.split('=')
```
Comment (charlesccychen) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: Let's not reuse `label`; how about `parts = label.split('=', 1)`?

File: sdks/python/apache_beam/options/pipeline_options.py (hunk 1 above)
Comment (aaltay) [inline sdks/python/apache_beam/options/pipeline_options.py]: > **charlesccychen** wrote:
> Is "labels" a general purpose Dataflow concept, or just for billing?


Removed billing.

It reads like a general GCP thing (https://cloud.google.com/resource-manager/docs/using-labels) and shown in the UI similar to pipeline options in a separate are with no mention of billing.

@zongweiz is this accurate?

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py (hunk 2 above)
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: > **charlesccychen** wrote:
> nit: period after


Done.

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py (hunk 3 above)
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: > **charlesccychen** wrote:
> Let's not reuse `label`; how about `parts = label.split('=', 1)`?


Done. Thank you.

**Pull request closed without merging**